relay-server/src/services/processor.rs

```diff
@@ -1699,6 +1698,7 @@ impl EnvelopeProcessorService {
         if_processing!(self.inner.config, {
             // Process profiles before extracting metrics, to make sure they are removed if they are invalid.
             let profile_id = profile::process(state, &self.inner.config);
+            profile::transfer_id(state, profile_id);
```

Comment:
I think we should merge this step with `profile::process` instead of having it a separate step now.

Call it from withing `profile::porcess` and make the `transfer_id` function private.

Has a side-effect of making the `cfg(feture = "processing")` dance obsolete.

Reply:
I'm going to leave this alone as I've put back the `transfer_id` call above due to dynamic sampling.